src/components/action-popover/action-popover.spec.tsx

```diff
@@ -180,6 +181,249 @@ describe("ActionPopover", () => {
+          Download PDF
+        </ActionPopoverItem>,
+        <ActionPopoverItem key="item-5" onClick={() => {}} submenu={submenu}>
+          Download CSV
+        </ActionPopoverItem>,
+        <ActionPopoverDivider key="item-6" />,
+        <ActionPopoverItem key="item-7" onClick={() => {}} submenu={submenu}>
+          Delete
+        </ActionPopoverItem>,
+        <ActionPopoverItem key="item-8" onClick={() => {}} submenu={submenu}>
+          Return Home
+        </ActionPopoverItem>,
+        null,
+        undefined,
+      ],
+      ...props,
+    };
+
+    renderer(
+      <ThemeProvider theme={mintTheme}>
+        <>
+          <input id="before" />
+          <ActionPopover {...defaultProps} {...props} />
+          <input id="after" />
+        </>
+      </ThemeProvider>
+    );
+  }
+
+  function renderActionPopoverWithSubmenusAndSomeIcons(
+    props = {},
+    renderer = mount
+  ) {
+    const submenu = (
+      <ActionPopoverMenu>
+        <ActionPopoverItem icon="bin" onClick={() => {}}>
+          Sub Menu 1
+        </ActionPopoverItem>
+        <ActionPopoverItem onClick={() => {}}>Sub Menu 2</ActionPopoverItem>
+        <ActionPopoverItem onClick={() => {}}>Sub Menu 3</ActionPopoverItem>
+        <ActionPopoverItem onClick={() => {}}>Sub Menu 4</ActionPopoverItem>
+      </ActionPopoverMenu>
+    );
+
+    const defaultProps = {
+      children: [
+        <ActionPopoverItem key="item-1" onClick={() => {}} submenu={submenu}>
+          Business
+        </ActionPopoverItem>,
+        <ActionPopoverItem key="item-2" onClick={() => {}} submenu={submenu}>
+          Email Invoice
+        </ActionPopoverItem>,
+        <ActionPopoverItem key="item-3" onClick={() => {}} submenu={submenu}>
+          Print Invoice
+        </ActionPopoverItem>,
+        <ActionPopoverItem key="item-4" onClick={() => {}} submenu={submenu}>
+          Download PDF
+        </ActionPopoverItem>,
+        <ActionPopoverItem
+          key="item-5"
+          icon="csv"
+          onClick={() => {}}
+          submenu={submenu}
+        >
+          Download CSV
+        </ActionPopoverItem>,
+        <ActionPopoverDivider key="item-6" />,
+        <ActionPopoverItem
+          key="item-7"
+          icon="delete"
+          onClick={() => {}}
+          submenu={submenu}
+        >
+          Delete
+        </ActionPopoverItem>,
+        <ActionPopoverItem
+          key="item-8"
+          icon="home"
+          onClick={() => {}}
+          submenu={submenu}
+        >
+          Return Home
+        </ActionPopoverItem>,
+        null,
+        undefined,
+      ],
+      ...props,
+    };
+
+    renderer(
+      <ThemeProvider theme={mintTheme}>
+        <>
+          <input id="before" />
+          <ActionPopover {...defaultProps} {...props} />
+          <input id="after" />
+        </>
+      </ThemeProvider>
+    );
+  }
+
+  function renderActionPopoverWithSubmenusAndIconsAndNoSubmenusAndIcons(
```

Comment:
```suggestion
  function renderActionPopoverWithVariableChildren(
```

**suggestion** (non-blocking): I don't know if I even like this name, there's probably something better. I don't mind the length of the original name at all, if it's descriptive, the problem I have is that it sounds contradictory - and I only discovered by asking you in person that it means that some children have submenu/icon and some don't. (Which is absolutely a case we should test, my comment is purely about the naming.)

Reply:
Yeah agree this name could be a bit shorter, you could even just go with `renderWithVariableChildren`